src/NServiceBus.Core/OpenTelemetry/Metrics/MessagingMetricsFeature.cs

```diff
@@ -22,6 +24,16 @@ protected internal override void Setup(FeatureConfigurationContext context)
             "Provides OpenTelemetry counters for message processing"
         );
         var criticalTimeMetrics = new CriticalTimeMetrics(queueName, discriminator);
-        context.Pipeline.OnReceivePipelineCompleted((pipeline, token) => criticalTimeMetrics.Record(pipeline, token));
+        var processingTimeMetrics = new ProcessingTimeMetrics(queueName, discriminator);
+        context.Pipeline.OnReceivePipelineCompleted((pipeline, token) =>
+        {
+            List<Task> tasks =
+            [
+                processingTimeMetrics.Record(pipeline, token),
+                criticalTimeMetrics.Record(pipeline, token)
+            ];
+
+            return Task.WhenAll(tasks);
+        });
```

Comment:
Do we need to allocate a list?
```suggestion
        context.Pipeline.OnReceivePipelineCompleted((pipeline, token) => Task.WhenAll(
            processingTimeMetrics.Record(pipeline, token),
            criticalTimeMetrics.Record(pipeline, token)
        );
```